### PR TITLE
Minor Table component fixes

### PR DESCRIPTION
**Pull request by kofi-clever**
- Reduce padding for inner cells to reduce cell spacing.
  **Before:**
screen shot 2016-08-26 at 1 51 06 am

  **After:**
screen shot 2016-08-26 at 1 51 19 am

- Handle data prop updates and update rows accordingly, while preserving
  sort state

Also, update the TableExample to demo updating the table with new data:
table-reload-demo

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -35,3 +35,6 @@ $(TESTS):
 	@echo "Running tests for $@"
 	@$(LINT) $@
 	@NODE_ENV=test $(MOCHA) $(MOCHA_OPTIONS) $@
+
+dev-server:
```
Comment (ipam73) [inline Makefile]: could you update the readme to use `make dev-server`
https://github.com/Clever/components/blob/master/README.md#contributing

Comment (kofi-clever) [inline Makefile]: thanks, good call - https://github.com/Clever/components/pull/48/commits/9e52fdda9d38810157003446c3c2165bab29ffdb

File: src/Table/Table.jsx [hunk 2]
```diff
@@ -37,13 +46,13 @@ export class Table extends Component {
...
 
-    this._sort({
+    this._sort(this.props.data, {
       columnIndex,
       direction: sortDirection.ASCENDING,
     });
   }
 
-  _sort(sortState) {
+  _sort(data, sortState) {
```
Comment (ipam73) [inline src/Table/Table.jsx]: Is there a reason to pass `data` if it's not being used here?

Comment (ipam73) [inline src/Table/Table.jsx]: edit:  ignore this comment - see that it's being used below

**Comment by ipam73** [comment]
@kofi-clever I'd refer to style changes to someone from design, but other than that LGTM

**Comment by kofi-clever** [comment]
@CarlNelson - meant to ask actually: could you confirm the cell padding for the table?
i have it at 20px by 20px right now, based on https://clever.atlassian.net/browse/DESIGN-56
does that look ok?

**Pull request merged**